<launch>
    <arg name="camera_id" default="/usb_cam"/>
    <arg name="image_src" default="/image_raw"/>
    
    <arg name="camera_info_src" default="$(find darknet_d)/config/camera_info.yaml"/>

    <arg name="object_species_file" default="$(find darknet_d)/data/coco.names"/>
    <arg name="network_definition_file" default="$(find darknet_d)/model/yolov4.cfg"/>
    <arg name="pretrained_model_file" default="$(find darknet_d)/model/yolov4.weights"/>

    <arg name="score_threshold" default="0.40"/>
    <arg name="nms_threshold" default="0.30"/>

    <node pkg="darknet_d" name="darknet_detect" type="darknet_d" output="screen">
        <param name="base_dir" type="str" value="$(find darknet_d)"/>
        <param name="image_topic_name" type="str" value="$(arg camera_id)$(arg image_src)"/>
        <param name="camera_info_dir" type="str" value="$(arg camera_info_src)"/>
        <param name="names_file" type="str" value="$(arg object_species_file)"/>
        <param name="cfg_file" type="str" value="$(arg network_definition_file)"/>
        <param name="weights_file" type="str" value="$(arg pretrained_model_file)"/>
        <param name="conf_thresh" type="double" value="$(arg score_threshold)"/>
    </node>

</launch>
